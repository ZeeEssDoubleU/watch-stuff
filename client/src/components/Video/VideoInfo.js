import React, { useState } from "react";
import { Button, Image } from "semantic-ui-react";
import Linkify from "react-linkify";

import "./VideoInfo.scss";

import { getFormattedDate } from "../../utils/format-time";
import { getAbbrevNumber } from "../../utils/format-number";

const VideoInfo = props => {
	const [collapsed, toggleCollapse] = useState(true);
	const channel = props.channel;
	const video = props.video;

	if (!video) return <div />;

	const channelIcon =
		"https://yt3.ggpht.com/a-/AAuE7mAWpJP0ftMStYJD1f_tGS-RlvHU2rk7zcxRbg=s240-mo-c-c0xffffffff-rj-k-no";
	const channelName = channel
		? channel.snippet.title
		: video.snippet.channelTitle;
	const publishDate = getFormattedDate(video.snippet.publishedAt);
	const description = video.snippet.description;
	const subCount = channel
		? getAbbrevNumber(channel.statistics.subscriberCount)
		: null;

	return (
		<div className="video-info-container">
			<Image
				className="channel-image"
				src={channelIcon}
				circular
			/>
			<div className="video-info">
				<div className="channel-name">{channelName}</div>
				<div className="videopublication-date">
					Published on {publishDate}
				</div>
			</div>
			<Button className="subscribe-button" color="youtube">
				SUBSCRIBE {subCount}
			</Button>
			<div className="video-description">
				<div className={collapsed ? "collapsed" : "expanded"}>
					<Linkify>{description}</Linkify>
				</div>
				<Button
					className="show-more-button"
					compact
					onClick={() => toggleCollapse(!collapsed)}>
					{collapsed ? "SHOW MORE" : "SHOW LESS"}
				</Button>
			</div>
		</div>
	);
};

export default VideoInfo;
