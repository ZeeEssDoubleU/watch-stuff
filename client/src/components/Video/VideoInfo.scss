@import '../../styles/shared.scss';
.video-info-container {
   display: grid;
   grid: auto auto / max-content auto 156px;
   align-items: center;
   grid-row-gap: 16px;
   .channel-image {
      margin-right: $avatar-margin;
      grid-area: 1/1 / 2/2;
      cursor: pointer;
   }
   .video-info {
      grid-area: 1/2 / 2/3;
      .channel-name {
         font-weight: 600;
         cursor: pointer;
      }
      .video-publication-date {
         font-size: 13px;
         color: #707070;
      }
   }
   .subscribe-button {
      grid-area: 1/3 / 2/4;
      &.ui.button {
         margin: 0;
      }
   }
   .video-description {
      grid-area: 2/2 / 3/4;
      max-width: 615px;
      white-space: pre-line;
      &>.ui.button {
         margin: 8px 0;
         padding-left: 0;
         padding-right: 0;
         background: none;
      }
      p {
         line-height: 2rem;
         margin-bottom: 0;
      }
      .collapsed {
         max-height: 4rem;
         overflow-y: hidden;
      }
      .expanded {
         max-height: none;
      }
   }
}