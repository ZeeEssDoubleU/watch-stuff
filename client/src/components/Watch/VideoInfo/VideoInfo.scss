.video-info-container {
   display: grid;
   grid: auto auto / max-content auto max-content;
   align-items: center;
   grid-row-gap: 16px;
   .channel-image {
      margin: 5px;
      grid-area: 1/1 / 2/2;
      cursor: pointer;
   }
   .video-info {
      grid-area: 1/2 / 2/3;
      .channel-name {
         font-weight: 600;
         cursor: pointer;
      }
      .video-publication-date {
         font-size: 13px;
         color: #707070;
      }
   }
   .subscribe-button {
      grid-area: 1/3 / 2/4;
      &.ui.button {
         margin: 0;
      }
   }
   .video-description {
      grid-area: 2/2 / 3/3;
      max-width: 615px;
      &>.ui.button {
         margin: 8px 0;
         padding-left: 0;
         padding-right: 0;
         background: none;
      }
      p {
         line-height: 1.4rem;
         margin-bottom: 0;
      }
      .collapsed {
         max-height: 2.8rem;
         overflow-y: hidden;
      }
      .expanded {
         max-height: none;
      }
   }
}