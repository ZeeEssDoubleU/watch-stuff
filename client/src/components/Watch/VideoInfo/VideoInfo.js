import React, { Component } from "react";
import PropTypes from "prop-types";
import { Divider } from "semantic-ui-react";

import "./VideoInfo.scss";

class VideoInfo extends Component {
	render() {
		return (
			<div className="video-info">
				Video Info here.
				<Divider />
			</div>
		);
	}
}

VideoInfo.propTypes = {};

export default VideoInfo;
