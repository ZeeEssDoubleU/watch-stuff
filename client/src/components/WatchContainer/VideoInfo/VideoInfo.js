import React, { Component } from "react";
import PropTypes from "prop-types";
import { Button, Divider, Image } from "semantic-ui-react";

import "./VideoInfo.scss";

class VideoInfo extends Component {
	constructor(props) {
		super(props);
		this.state = {
			collapsed: true,
		};
	}

	onToggleCollapseButtonClick = () => {
		this.setState({
			collapsed: !this.state.collapsed,
		});
	};

	render() {
		let visibility = "collapsed";
		let buttonText = "SHOW MORE";
		if (!this.state.collapsed) {
			visibility = "expanded";
			buttonText = "SHOW LESS";
		}

		let description = (
			<p>
				Lorem ipsum dolor, sit amet consectetur adipisicing elit. Autem quos, magnam impedit
				vero hic inventore, numquam perspiciatis veritatis obcaecati aut assumenda rerum. Atque
				ratione similique numquam excepturi provident, commodi fugiat dolores accusamus quas
				cupiditate dolorum facilis iure aliquam nesciunt exercitationem totam pariatur deserunt
				nemo cum in. Eum voluptates mollitia quisquam eos repellendus sapiente ratione
				accusantium tempore sit dolorum dolorem est aut dolores nesciunt, ipsum voluptas tempora
				sequi praesentium obcaecati amet! Nam nostrum deleniti a eius, recusandae dolorum? Eius
				tenetur repudiandae debitis incidunt aliquid harum, deserunt, illum tempore,
				voluptatibus fuga aspernatur eaque consequuntur magnam perferendis consectetur earum
				veritatis nisi ut?
			</p>
		);

		return (
			<div className="video-info-container">
				<Image className="channel-image" src="http://via.placeholder.com/48x48" circular />
				<div className="video-info">
					<div className="channel-name">Channel Name</div>
					<div className="videopublication-date">Published on Jun 25, 2019</div>
				</div>
				<Button className="subscribe-button" color="youtube">
					SUBSCRIBE 523K
				</Button>
				<div className="video-description">
					<div className={visibility}>{description}</div>
					<Button
						className="show-more-button"
						compact
						onClick={this.onToggleCollapseButtonClick}>
						{buttonText}
					</Button>
				</div>
			</div>
		);
	}
}

VideoInfo.propTypes = {};

export default VideoInfo;
