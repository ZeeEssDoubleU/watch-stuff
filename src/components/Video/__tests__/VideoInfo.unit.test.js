import React from "react";
import { shallow } from "enzyme";

import VideoInfo from "../VideoInfo";
import watchDetails from "./props/watchDetails";

describe("VideoInfo", () => {
	test("renders without props", () => {
		const wrapper = shallow(<VideoInfo />);
		expect(wrapper).toMatchSnapshot();
	});
	test("renders collapsed with video prop", () => {
		const wrapper = shallow(<VideoInfo video={watchDetails} />);
		expect(wrapper).toMatchSnapshot();
	});
	test("renders expanded with video prop", () => {
		const wrapper = shallow(<VideoInfo video={watchDetails} />);
		wrapper.setState({ collapsed: false });
		expect(wrapper).toMatchSnapshot();
	});
});
