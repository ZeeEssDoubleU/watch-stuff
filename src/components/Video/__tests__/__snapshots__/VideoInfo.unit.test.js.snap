// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInfo renders collapsed with video prop 1`] = `
<div
  className="video-info-container"
>
  <Image
    as="img"
    circular={true}
    className="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      PlayOverwatch
    </div>
    <div
      className="videopublication-date"
    >
      Published on 
      Nov 2, 2018
    </div>
  </div>
  <Button
    as="button"
    className="subscribe-button"
    color="youtube"
  >
    SUBSCRIBE 523K
  </Button>
  <div
    className="video-description"
  >
    <div
      className="collapsed"
    >
      <Linkify
        className="Linkify"
        component="a"
        properties={Object {}}
      >
        An explosive train robbery gives McCree the chance to settle some unfinished business with a few former associates in our latest animated short: Reunion! 

Learn more about McCree: https://playoverwatch.com/en-us/heroes/mccree/

Begin your watch: http://buyoverwatch.com
Like us on Facebook: http://www.facebook.com/playoverwatch
Follow us on Twitter: http://www.twitter.com/playoverwatch
Join us on Instagram: https://www.instagram.com/playoverwatch
      </Linkify>
    </div>
    <Button
      as="button"
      className="show-more-button"
      compact={true}
      onClick={[Function]}
    >
      SHOW MORE
    </Button>
  </div>
</div>
`;

exports[`VideoInfo renders expanded with video prop 1`] = `
<div
  className="video-info-container"
>
  <Image
    as="img"
    circular={true}
    className="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      PlayOverwatch
    </div>
    <div
      className="videopublication-date"
    >
      Published on 
      Nov 2, 2018
    </div>
  </div>
  <Button
    as="button"
    className="subscribe-button"
    color="youtube"
  >
    SUBSCRIBE 523K
  </Button>
  <div
    className="video-description"
  >
    <div
      className="expanded"
    >
      <Linkify
        className="Linkify"
        component="a"
        properties={Object {}}
      >
        An explosive train robbery gives McCree the chance to settle some unfinished business with a few former associates in our latest animated short: Reunion! 

Learn more about McCree: https://playoverwatch.com/en-us/heroes/mccree/

Begin your watch: http://buyoverwatch.com
Like us on Facebook: http://www.facebook.com/playoverwatch
Follow us on Twitter: http://www.twitter.com/playoverwatch
Join us on Instagram: https://www.instagram.com/playoverwatch
      </Linkify>
    </div>
    <Button
      as="button"
      className="show-more-button"
      compact={true}
      onClick={[Function]}
    >
      SHOW LESS
    </Button>
  </div>
</div>
`;

exports[`VideoInfo renders without props 1`] = `<div />`;
